### PR TITLE
Portals mostly working

### PR DESCRIPTION
**Pull request by leem42**
Scsss - Added margin to buttons on navbar, modified icon scss on footer/header, removed button coloring through scss
Removed Explore Component --- made it a custom-component
Added component for query wrapper that sets up the stacked bar chart
Reworked types in portal-types to use interface, also simplified route to either base route or nested route
Added components that wrap around explore buttons
Added configs for NF and AMP-AD

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 node_modules/
 build/
-app-template/src/config/**
\ No newline at end of file
+app-template/src/config/**
+undoLink.sh
```
Comment (leem42) [inline .gitignore]: add these to the readme!

File: app-template/src/tests/portal-components/ExploreButtonControlWrapper.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      {
+        name: 'Custom Route 1'
+      }
+    ]
+  }
+
+  it('renders correctly', () => {
+    const component = mount(<MemoryRouter><ExploreButtonControlWrapper {...props} /></MemoryRouter>)
+    // check its defined
+    expect(component).toBeDefined()
```
Comment (leem42) [inline app-template/src/tests/portal-components/ExploreButtonControlWrapper.test.tsx]: check color of the button and name of route!

File: app-template/src/tests/portal-components/HomeButtonControlWrapper.test.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import { mount } from 'enzyme'
+import { MemoryRouter } from 'react-router'
+import CardContainerLogic from 'synapse-react-client/dist/containers/CardContainerLogic'
+import HomeButtonControlWrapper, { HomeButtonControlWrapperProps } from '../../portal-components/HomeButtonControlWrapper'
+
+describe('ExploreButtonControl works', () => {
```
Comment (leem42) [inline app-template/src/tests/portal-components/HomeButtonControlWrapper.test.tsx]: change description, check color and name of route!

File: app-template/src/types/portal-config.d.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-import { StackedBarChartControlProps } from "../custom-components/StackedBarChartControl";
+import { CardContainerLogicProps } from 'synapse-react-client/dist/containers/CardContainerLogic'
+import { MarkdownSynapseProps } from 'synapse-react-client/dist/containers/MarkdownSynapse'
+import { StackedBarChartProps } from 'synapse-react-client/dist/containers/StackedBarChart'
+import { QueryWrapperMenuProps } from 'synapse-react-client/dist/containers/QueryWrapperMenu'
+import { QueryBundleRequest } from 'synapse-react-client/dist/utils/jsonResponses/Table/QueryBundleRequest'
+import { QueryResultBundle } from 'synapse-react-client/dist/utils/jsonResponses/Table/QueryResultBundle'
+import { HomeButtonControlProps } from '../portal-components/HomeButtonControlWrapper'
+import { ExploreButtonControlProps } from '../portal-components/ExploreButtonControlWrapper'
+import { QueryWrapperProps } from 'synapse-react-client/dist/containers/QueryWrapper'
```
Comment (leem42) [inline app-template/src/types/portal-config.d.ts]: does not exist yet, still needs to get exported from SRC, will break build if staging is run for any portal

**Comment by jay-hodgson** [comment]
and build common js for SRC

File: .gitignore (hunk 1 above)
Comment (jay-hodgson) [reply .gitignore]: and check them in (if they are a useful dev tool)

**Pull request merged**